pkg/util/ipset/ipset.go

```diff
@@ -309,17 +423,85 @@ func getIPSetVersionString(exec utilexec.Interface) (string, error) {
 	return match[0], nil
 }
 
-func validatePortRange(portRange string) bool {
+// checks if port range is valid. The begin port number is not necessarily less than
+// end port number - ipset util can accept it.  It means both 1-100 and 100-1 are valid.
+func validatePortRange(portRange string) (bool, error) {
 	strs := strings.Split(portRange, "-")
 	if len(strs) != 2 {
-		return false
+		return false, fmt.Errorf("port range should be in the format of `a-b`")
 	}
 	for i := range strs {
-		if _, err := strconv.Atoi(strs[i]); err != nil {
-			return false
+		num, err := strconv.Atoi(strs[i])
+		if err != nil {
+			return false, fmt.Errorf("%s is not an integer number", strs[i])
+		}
+		if num < 0 {
+			return false, fmt.Errorf("port number %d should be >=0", num)
+		}
+	}
+	return true, nil
+}
+
+// checks if the given ipset type is valid.
+func validateIPSetType(set Type) (bool, error) {
+	for _, valid := range ValidIPSetTypes {
+		if set == valid {
+			return true, nil
+		}
+	}
+	return false, fmt.Errorf("unsupported set type: %s", set)
+}
+
+// checks if given hash family is supported in ipset
+func validateHashFamily(family string) (bool, error) {
```

Comment:
I don't see why this function should return an error...

Reply:
I return an error just for the sake of telling caller the reason... 

Do you think logging the error is a good idea?